src/to_vec.jl

```diff
@@ -162,11 +162,24 @@ function to_vec(x::Cholesky)
     return x_vec, Cholesky_from_vec
 end
 
-function to_vec(x::S) where {S <: Union{LinearAlgebra.QRCompactWYQ, LinearAlgebra.QRCompactWY}}
-    x_vec, back = to_vec([x.factors, x.T])
+function to_vec(x::S) where {U, S <: Union{LinearAlgebra.QRCompactWYQ{U}, LinearAlgebra.QRCompactWY{U}}}
```

Comment:
Are you sure this is valid for `QRCompactWYQ`? If it is, it should at least be tested.

Reply:
`QRCompactWYQ` is represented with the same `T` and `factors` matrices than `QRCompactWY`, yes. I added the tests.

Reply:
Perfect, thanks!